xarray/core/dataset.py

```diff
@@ -1204,9 +1277,22 @@ def sel(self, method=None, tolerance=None, drop=False, **indexers):
         Dataset.isel_points
         DataArray.sel
         """
+        from .dataarray import DataArray
+
+        new_coords = {k: v._coords for k, v in indexers.items()
+                      if isinstance(v, DataArray)}
+
+        indexers = {k: v.variable if isinstance(v, DataArray) else v
+                    for k, v in indexers.items()}
+
         pos_indexers, new_indexes = indexing.remap_label_indexers(
             self, indexers, method=method, tolerance=tolerance
         )
+        # attach indexer's coordinate to pos_indexers
+        for k, v in new_coords.items():
+            if isinstance(pos_indexers[k], Variable):
+                pos_indexers[k] = DataArray(pos_indexers[k], coords=v,
+                                            dims=pos_indexers[k].dims)
```

Comment:
Again, consider going back to the original `indexers` arguments to get `coords` and `dims`.

Reply:
Done